OpenMage/magento-lts
# Remove deprecated Admin user multirole functionality

### Description (*)

This PR aims to get rid of all the remnant code of multi-role functionality for Admin user. This functionality was deprecated ages ago and it's just tech debt at this point. Currently, only one Role can be assigned to an Admin user.

This is not meant to introduce any major breaking changes, although I did a change in a template file in adminhtml, but that's because I doubt someone will have that file overridden.

#### Todo

- [ ] Consolidate `Mage_Admin_Model_Resource_User::saveRelations()` and `Mage_Admin_Model_Resource_User::add()` methods, both used to add a role to a user.
- [ ] Deprecate methods that operate on array of roles (e.g. `Mage_Admin_Model_Resource_User::getRoles`), and replace them with methods that work with a single role.

### Manual testing scenarios (*)

1. Assigning a Role to an Admin user from user edit page should work as usual.
2. Assigning a Role to an Admin user from role edit page (Role Users tab) should work as usual.

### Contribution checklist (*)
